src/codemagic/models/xcodebuild.py

```diff
@@ -55,8 +55,13 @@ def _log_process(self, xcodebuild_cli_process: Optional[XcodebuildCliProcess]):
         with self.logs_path.open('a') as fd, xcodebuild_cli_process.log_path.open('r') as process_logs:
             fd.write(f'>>> {xcodebuild_cli_process.safe_form}')
             fd.write('\n\n')
-            for chunk in process_logs.read(8192):
+
+            chunk = process_logs.read(8192)
+            while chunk:
                 fd.write(chunk)
+                chunk = process_logs.read(8192)
+            fd.write(process_logs.read())
```

Comment:
Why is this necessary?

Reply:
Possible race-condition when the process is still writing logs to the file. Happens sometimes if underlying xcodebuild process exits unexpectedly and buffers are not properly flushed.

Reply:
I recommend adding a comment

```suggestion
            # do an extra read in case xcodebuild exits unexpectedly and did not flush last buffer
            fd.write(process_logs.read()) 
```